racket/collects/racket/math.rkt

```diff
@@ -58,6 +58,17 @@
           [(real? z)
            (let loop ([z z])
              (cond [(z . < . 0) (- (loop (- z)))]
+                   [(< z .13)
+                    ;; Taylor expansion near 0 to avoid cancellation
+                    ;~ z+z^3/3!+z^5/5!+...
+                    (define z^2 (* z z))
+                    (define-values (s z^i)
+                      (for/fold ([s z]
+                                 [z^i (* z z^2)])
```

Comment:
Instead of using `define-values`, you can use:

`(for/fold ([s ....] [z^i ....] #:result s) (....) ....)`

Reply:
ah thanks,
now I wish I knew this before...